### PR TITLE
feat: send emails on success/failure

### PR DESCRIPTION
**Pull request by kormide**
This adds the capability to send emails to the releaser. Notifications that are actually helpful will go into another PR.

### CODE REVIEW

File: src/application/notifications.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { User } from "../domain/user.js";
+import { Authentication, EmailClient } from "../infrastructure/email.js";
+import { SecretsClient } from "../infrastructure/secrets.js";
+
+export class NotificationsService {
+  private readonly sender = process.env.NOTIFICATIONS_EMAIL;
```
Comment (alexeagle) [inline src/application/notifications.ts:6]: nit: error handling if it's unset?

File: src/application/notifications.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { User } from "../domain/user.js";
+import { Authentication, EmailClient } from "../infrastructure/email.js";
```
Comment (alexeagle) [inline src/application/notifications.ts:2]: how come the imports have explicit `.js` suffixes? is that required for ESM loading or something?

File: src/application/notifications.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  }
+
+  public async notifySuccess(recipient: User): Promise<void> {
+    await this.populateAuth();
+    await this.emailClient.sendEmail(
+      "smtp.gmail.com",
+      recipient.email,
+      this.sender,
+      "Publish to BCR [no-reply]",
+      "Successfully published pull request to the Bazel Central Registry."
```
Comment (alexeagle) [inline src/application/notifications.ts:42]: Link to the PR?

File: src/domain/user.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export class User {
```
Comment (alexeagle) [inline src/domain/user.ts:1]: nit: seems overly OOP to have a class for something that could just be a well-typed object like 
```
export interface User {
  username: string;
  email string;
}
```

**Review by alexeagle — APPROVED** [review]

File: src/domain/user.ts (hunk 4 above)
Comment (kormide) [reply src/domain/user.ts]: Good call. Fixed.

File: src/application/notifications.ts (hunk 3 above)
Comment (kormide) [reply src/application/notifications.ts:42]: This is just placeholder comment. I'm planning to do a pass on the email content in another PR.

File: src/application/notifications.ts (hunk 2 above)
Comment (kormide) [reply src/application/notifications.ts:2]: Yeah the node runtime on Google Cloud complains if we don't have these.

File: src/application/notifications.ts (hunk 1 above)
Comment (kormide) [reply src/application/notifications.ts]: Fixed

**Comment by kormide** [comment]
Updated this to use any SMTP service rather than assuming Gmail/Oauth.

**Pull request merged**